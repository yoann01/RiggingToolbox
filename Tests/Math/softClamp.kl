
require RiggingToolbox;

operator entry(){
   
    Scalar minVal = 3.0;
    Vec2 minValSoftening(2.0, 3.0);
    Scalar maxVal = 15.0;
    Vec2 maxValSoftening(3.0, 5.0);

    for(Integer i=0; i<28; i++){
        Scalar inval = Scalar(i);
        Scalar outval = softClamp(inval, minVal, minValSoftening, maxVal, maxValSoftening);
        report("inval:" + inval + " outval:" + outval);
    }
}

