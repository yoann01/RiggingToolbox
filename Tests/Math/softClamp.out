inval:+0.0 outval:+3.0
inval:+1.0 outval:+3.0
inval:+2.0 outval:+3.6
inval:+3.0 outval:+4.2
inval:+4.0 outval:+4.8
inval:+5.0 outval:+5.4
inval:+6.0 outval:+6.0
inval:+7.0 outval:+7.0
inval:+8.0 outval:+8.0
inval:+9.0 outval:+9.0
inval:+10.0 outval:+10.0
inval:+11.0 outval:+11.0
inval:+12.0 outval:+12.0
inval:+13.0 outval:+12.375
inval:+14.0 outval:+12.75
inval:+15.0 outval:+13.125
inval:+16.0 outval:+13.5
inval:+17.0 outval:+13.875
inval:+18.0 outval:+14.25
inval:+19.0 outval:+14.625
inval:+20.0 outval:+15.0
inval:+21.0 outval:+15.0
inval:+22.0 outval:+15.0
inval:+23.0 outval:+15.0
inval:+24.0 outval:+15.0
inval:+25.0 outval:+15.0
inval:+26.0 outval:+15.0
inval:+27.0 outval:+15.0
